src/client/Microsoft.Identity.Client/Utils/JsonHelper.cs

```diff
@@ -31,14 +32,19 @@ internal static string SerializeToJson<T>(T toEncode)
 #endif
         }
 
+#if SUPPORTS_SYSTEM_TEXT_JSON
+        [RequiresUnreferencedCode("Calls System.Text.Json.JsonSerializer.Deserialize(String, Type, JsonSerializerOptions)")]
+#endif
         internal static T DeserializeFromJson<T>(string json)
         {
             if (string.IsNullOrEmpty(json))
             {
                 return default;
             }
 #if SUPPORTS_SYSTEM_TEXT_JSON
-            return (T)JsonSerializer.Deserialize(json, typeof(T), MsalJsonSerializerContext.Custom);
+            //Workaround for https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/3892
+            //This also requires the RequiresUnreferencedCode attribute to maintain functionality during trimming.
+            return (T)JsonSerializer.Deserialize(json, typeof(T), MsalJsonSerializerOptions.Options);
```

Comment:
Found a potential solution.
So there are two ways to set serialization/deserialization behavior in `System.Test.Json` (there may be others)
- `JsonSerializerContext` 
- `JsonSerializerOptions`. 

It seems like tergetting .NET 7 causes the 
`object? Deserialize(string json, Type returnType, JsonSerializerContext context)` api to ignore the configuration in the `JsonSerializerContext`. 

However, using `object? Deserialize(string json, Type returnType, JsonSerializerOptions? options = null)` and setting `JsonSerializerOptions` seems to work.

Since this isn't a documented change, perhaps I should open an issue on the .NET 7 repo? I doubt they intended to ignore `JsonSerializerContext` when deserializing. @pmaytak, @bgavrilMS, @SameerK-MSFT?

Reply:
Yea, I think opening an issue there would be good. You can also search through 7.0.x [milestones](https://github.com/dotnet/runtime/milestones?state=closed) or issues, maybe this was already reported or fixed. Otherwise, getting some clarify would be good.

Reply:
Not honoring JsonSerializerContext would be a serious bug. It would be good to check if it already exists.

Reply:
Changing solution to use attribute on serializable values